sw/host/rom_ext_image_tools/signer/image/src/image.rs

```diff
@@ -31,51 +38,97 @@ impl RawImage {
+    /// Updates the fields from the configuration file.
     ///
-    /// This function updates the image manifest data with values parsed
-    /// from configuration file.
-    pub fn update_generic_fields(&mut self, config: &ParsedConfig) {
-        // TODO checks to make sure that the config values (strings) are not
-        // bigger than the actual field size.
+    /// This function updates the image manifest data with values parsed from
+    /// the configuration file (known ahead of time). Some of the other fields
+    /// like signature key public exponent and modulus, are obtained at
+    /// run-time.
+    pub fn update_static_fields(&mut self, config: &ParsedConfig) {
+        // TODO: checks to make sure that the config values (strings) are not
+        //       bigger than the actual field size.
 
-        let mut myclosure = |value, offset| {
+        let mut update = |value, offset| {
             let bytes = str_to_vec_u8(value);
-            let data = &mut self.data;
-            let begin = offset as usize;
-            let end = begin + bytes.len();
-            data.splice(begin..end, bytes.iter().cloned());
+            self.update_field(&bytes, offset);
         };
 
-        myclosure(
+        update(
             &config.manifest_identifier,
             manifest::ROM_EXT_MANIFEST_IDENTIFIER_OFFSET,
         );
-        myclosure(&config.image_length, manifest::ROM_EXT_IMAGE_LENGTH_OFFSET);
-        myclosure(
+        update(
             &config.image_version,
             manifest::ROM_EXT_IMAGE_VERSION_OFFSET,
         );
-        myclosure(
-            &config.image_timestamp,
-            manifest::ROM_EXT_IMAGE_TIMESTAMP_OFFSET,
+
+        update(
+            &config.extensions[0].offset,
+            manifest::ROM_EXT_EXTENSION0_OFFSET_OFFSET,
         );
-        myclosure(
-            &config.extension0_checksum,
+        update(
+            &config.extensions[0].checksum,
             manifest::ROM_EXT_EXTENSION0_CHECKSUM_OFFSET,
         );
-        myclosure(
-            &config.extension1_checksum,
+        update(
+            &config.extensions[1].offset,
+            manifest::ROM_EXT_EXTENSION1_OFFSET_OFFSET,
+        );
+        update(
+            &config.extensions[1].checksum,
             manifest::ROM_EXT_EXTENSION1_CHECKSUM_OFFSET,
         );
-        myclosure(
-            &config.extension2_checksum,
+        update(
+            &config.extensions[2].offset,
+            manifest::ROM_EXT_EXTENSION2_OFFSET_OFFSET,
+        );
+        update(
+            &config.extensions[2].checksum,
             manifest::ROM_EXT_EXTENSION2_CHECKSUM_OFFSET,
         );
-        myclosure(
-            &config.extension3_checksum,
+        update(
+            &config.extensions[3].offset,
+            manifest::ROM_EXT_EXTENSION3_OFFSET_OFFSET,
+        );
+        update(
+            &config.extensions[3].checksum,
             manifest::ROM_EXT_EXTENSION3_CHECKSUM_OFFSET,
         );
```

Comment:
This kind of duplication is a bit error-prone.

```rs
let offsets = [
  (ROM_EXT_EXTENSION0_OFFSET, ROM_EXT_EXTENSION0_CHECKSUM_OFFSET),
  // ...
];
for (extn_offset, checksum_offset) in offsets.iter().enumerate() {
  self.update_field(&str_to_vec_u8(&config.extensions[i].offset), *extn_offset);
  self.update_field(&str_to_vec_u8(&config.checksums[i].offset), *checksum_offset);
}
```
If you do this, you can get rid of the lambda at the top of this stanza.